### PR TITLE
Hotfix profile photo

### PR DESCRIPTION
**Pull request by lipob**
### What this PR does (required):
- Update DB user and  profile models. Now the profilePhoto property is in Profile model.
- Upload and Delete profile photo features tested and working fine.

### CODE REVIEW

File: server/controllers/auth.js [hunk 1]
```diff
@@ -96,6 +96,7 @@ exports.loginUser = asyncHandler(async (req, res, next) => {
 // @access Private
 exports.loadUser = asyncHandler(async (req, res, next) => {
   const user = await User.findById(req.user.id);
+  const userProfile = await Profile.findById(req.user.id);
```
Comment (ezeilo-su) [inline server/controllers/auth.js:99]: Would be good to validate the user id in the `req` object first before you go ahead and use it. If it's not supplied, then you return an error status to the client.

File: server/controllers/profile.js [hunk 2]
```diff
@@ -114,4 +116,39 @@ exports.findAllProfiles = asyncHandler(async (req, res, next) => {
       }
 
     res.status(200).json({ profiles: profiles });
```
Comment (ezeilo-su) [inline server/controllers/profile.js:118]: This should be one profile, `profile`. You don't have `profiles` defined!

Comment (ezeilo-su) [inline server/controllers/profile.js:118]: You can also consider using object property shorthand; just `{ profile }`

File: server/controllers/profile.js [hunk 3]
```diff
@@ -114,4 +116,39 @@ exports.findAllProfiles = asyncHandler(async (req, res, next) => {
...
-})
\ No newline at end of file
+})
+
+const deletePhoto = async (user) => {
+  await cloudinary.uploader.destroy(user.profilePhoto.publicId);
```
Comment (ezeilo-su) [inline server/controllers/profile.js:122]: Validate the request first so you can return a bad request status to the client if the required fields are missing.

File: server/controllers/profile.js [hunk 4]
```diff
@@ -114,4 +116,39 @@ exports.findAllProfiles = asyncHandler(async (req, res, next) => {
...
+  user.profilePhoto = null;
+};
+
+exports.uploadProfilePhoto = asyncHandler(async (req, res, next) => {
+  const image = req.file;
+  const user = await Profile.findById(req.user.id);
```
Comment (ezeilo-su) [inline server/controllers/profile.js:128]: Always validate the request first so you can return a bad request status to the client if the required fields are missing.

File: server/controllers/profile.js [hunk 5]
```diff
@@ -114,4 +116,39 @@ exports.findAllProfiles = asyncHandler(async (req, res, next) => {
...
+    url: image.path,
+    publicId: image.filename,
+  };
+  user.save();
+
+  res.sendStatus(200);
+});
+
+exports.deleteProfilePhoto = asyncHandler(async (req, res, next) => {
+  const user = await Profile.findById(req.user.id);
```
Comment (ezeilo-su) [inline server/controllers/profile.js:145]: Always validate the request first so you can return a bad request status to the client if the required fields are missing.

File: server/controllers/user.js [hunk 6]
```diff
@@ -23,36 +24,36 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
   res.status(200).json({ users: users });
```
Comment (ezeilo-su) [inline server/controllers/user.js:24]: You may consider using object property shorthand; just `{ users }`

File: server/db.js [hunk 7]
```diff
@@ -9,14 +9,14 @@ const connectDB = async () => {
...
-    });
-  }
+  // if (process.env.NODE_ENV !== 'production' && mongoose.connection.readyState === 1) {
+  //   console.log('Seeding database...');
+  //   const user = await User.create({
+  //     username: 'demo',
+  //     email: 'demo@email.com',
+  //     password: 'password',
+  //   });
+  // }
```
Comment (ezeilo-su) [inline server/db.js:19]: Consider removing unused codes.

File: server/controllers/user.js [hunk 8]
```diff
@@ -23,36 +24,36 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
...
+//   const user = await Profile.findById(req.user.id);
+//   if (!user) {
+//     return res.sendStatus(404);
+//   }
+
+//   await deletePhoto(user);
+//   user.save();
+
+//   res.sendStatus(200);
+// });
```
Comment (ezeilo-su) [inline server/controllers/user.js:59]: Consider removing unused codes.

File: server/routes/user.js [hunk 9]
```diff
@@ -1,17 +1,17 @@
 const express = require("express");
 const router = express.Router();
 const protect = require("../middleware/auth");
-const { multerUploads } = require("../utils/multer");
+// const { multerUploads } = require("../utils/multer");
```
Comment (ezeilo-su) [inline server/routes/user.js:4]: Consider removing unused codes.

File: server/routes/user.js [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
 
 router.route("/").get(protect, searchUsers);
-router
-  .route("/upload/profile-photo")
-  .post(protect, multerUploads.single("image"), uploadProfilePhoto);
-router.route("/profile-photo").delete(protect, deleteProfilePhoto);
+// router
+//   .route("/upload/profile-photo")
+//   .post(protect, multerUploads.single("image"), uploadProfilePhoto);
+// router.route("/profile-photo").delete(protect, deleteProfilePhoto);
```
Comment (ezeilo-su) [inline server/routes/user.js:15]: Consider removing unused codes.

**Review by ezeilo-su — COMMENTED** [review]
Good work on this one. But you need the read the comments I left.

File: server/controllers/auth.js (hunk 1 above)
Comment (lipob) [reply server/controllers/auth.js:99]: Hi Sunday. This route uses a protect middleware to decode the token and send the req.user data.

File: server/controllers/profile.js (hunk 3 above)
Comment (lipob) [reply server/controllers/profile.js:122]: done

File: server/controllers/auth.js (hunk 1 above)
Comment (ezeilo-su) [reply server/controllers/auth.js:99]: OK, @lipob. That means that validation has already taken place in the middleware. Thanks for letting me know this!

**Comment by lipob** [comment]
Fixed all suggestions. Waiting for aproval to merge. Thanks

File: server/controllers/auth.js (hunk 1 above)
Comment (lipob) [reply server/controllers/auth.js:99]: You're welcome.

File: server/controllers/profile.js (hunk 2 above)
Comment (lipob) [reply server/controllers/profile.js:118]: I don't know exactly how this function works, cause I didn't create it.

File: server/bin/www [hunk 11]
```diff
@@ -51,7 +51,7 @@ const authSocket = async (socket, next) => {
   return next();
 };
 
-io.use(authSocket);
+// io.use(authSocket);
```
Comment (moffatethan) [inline server/bin/www:54]: Remove?

Comment (lipob) [reply server/bin/www:54]: Can I remove it from my branch without any problems?

Comment (moffatethan) [reply server/bin/www:54]: The auth socket work hasn't been pushed in yet so no problem if you remove this

Comment (moffatethan) [reply server/bin/www:54]: We shouldn't even have this code yet in your branch 🤔

Comment (lipob) [reply server/bin/www:54]: Ok. I've just removed it. Thanks

**Review by moffatethan — APPROVED** [review]

**Pull request merged**